examples/hello_out_dir/BUILD

```diff
@@ -0,0 +1,26 @@
+package(default_visibility = ["//visibility:public"])
+
+load(
+    "@io_bazel_rules_rust//rust:rust.bzl",
+    "rust_binary",
+    "rust_library",
+    "rust_test",
+)
+
+rust_library(
+    name = "hello_out_dir",
+    srcs = ["src/lib.rs"],
+    out_dir_tar = ":repacked_srcs.tar.gz",
```

Comment:
can you really only ever have one of these for a target? Are there no cases where you might have more than one?

Reply:
It isn't possible to have more than one with Cargo -- and i didn't *expect* anyone to use this outside that context. I probably lack imagination on this point.

I don't think it would be much trouble to facilitate having more than one here but I'm not sure what the best way to handle files within the tars with the same name.

Reply:
we could fail the build if they collide, or just untar in order taking the rightmost. I would probably prefer failing I guess.... but if you think no one needs, just a comment.

Reply:
Added a note to the readme about this. I believe this sort of use case could be supported on the user side by unzipping several archives and merging them, which lets a user choose the best way to solve this problem.

(also it's a cop out on my end :stuck_out_tongue:)

Definitely willing to revisit this if an issue is opened or I missed a common use case!